src/resources/filters/quarto-post/email.lua

```diff
@@ -0,0 +1,423 @@
+      image_tbl[cid] = img_el.src
+      img_el.src = "cid:" .. cid
+      count = count + 1
+      return img_el
+
+    end})
+
+    email_html = extract_email_div_str(render_div_cid)
+
+    --[[
+    Render of HTML email message body for Connect: `email_html_preview`
+
+    We are keeping a render of the email HTML for previewing purposes (if the option
+    is taken to do so); here, the HTML is self-contained where the image tags contain
+    base64-encoded data
+    ]]
+
+    local render_div_base64 = quarto._quarto.ast.walk(div, {Image = function(img_el)
+
+      local image_file = io.open(img_el.src, "rb")
+
+      if type(image_file) == "userdata" then
+        image_data = image_file:read("*all")
+        image_file:close()
+      end
+
+      local encoded_data = quarto.base64.encode(image_data)
+      local file_extension = get_file_extension(img_el.src)
+      local base64_str = "data:image/" .. file_extension .. ";base64," .. encoded_data
+      img_el.src = base64_str
+      return img_el
+
+    end})
+
+    email_html_preview = extract_email_div_str(render_div_base64)
+
+    -- Remove the the `.email` div so it doesn't appear in the main report document
+    return {}
+
+  end
+end
+
+-- Function to extract the rendered HTML from a Div of class 'email'
+function extract_email_div_str(doc)
+  return pandoc.write(pandoc.Pandoc( {doc} ), "html")
+end
+
+function process_document(doc)
+
+  -- Get the current date and time
+  local connect_date_time = os.date("%Y-%m-%d %H:%M:%S")
+
+  -- Use Connect environment variables to get URLs for the email footer section
+  -- If any of these are nil, a portion of the email footer won't be rendered
+  local connect_report_rendering_url = os.getenv("RSC_REPORT_RENDERING_URL")
+  local connect_report_url = os.getenv("RSC_REPORT_URL")
+  local connect_report_subscription_url = os.getenv("RSC_REPORT_SUBSCRIPTION_URL")
+
+  -- The following regexes remove the surrounding <div> from the HTML text
+  email_html = string.gsub(email_html, "^<div class=\"email\">", '')
+  email_html = string.gsub(email_html, "</div>$", '')
+
+  -- Use the Connect email template components along with the `email_html` and
+  -- `email_html_preview` objects to generate the email message body for Connect
+  -- and the email HTML file (as a local preview)
+  
+  html_email_body = generate_html_email_from_template(
+    email_html,
+    connect_date_time,
+    connect_report_rendering_url,
+    connect_report_url,
+    connect_report_subscription_url
+  )
+
+  html_preview_body = generate_html_email_from_template(
+    email_html_preview,
+    connect_date_time,
+    connect_report_rendering_url,
+    connect_report_url,
+    connect_report_subscription_url
+  )
+
+  -- For each of the <img> tags we need to create a Base64-encoded representation
+  -- of the image and place that into the table `email_images` (keyed by `cid`)
+
+  local image_data = nil
+
+  for cid, img in pairs(image_tbl) do
+    if (true) then
```

Comment:
Guessing this might just be cruft...

Reply:
I removed the `if (true) then ...` conditional. Thanks!